scalding-spark/src/main/scala/com/twitter/scalding/spark_backend/Op.scala

```diff
@@ -175,11 +175,13 @@ object Op extends Serializable {
             .groupBy(_._1)
             .map { case (k, vs) => (k, vs.map(_._2)) }
 
+          val bcastMap = ctx.broadcast(rightMap)
```

Comment:
you can/probably should destroy the broadcast when your done with it... not sure if you can easily know that here though...

(I've only found one case where its caused problems not destroying them, they usually are eventually GC'd anyway )

Reply:
yeah, adding clean ups it a bit challenging, I was thinking of punting for now.

Basically, the `Op[A]` would need to return ` Future[(RDD[_ <: A], () => Future[Unit])]` or something so the right hand function is a cleanup on finish?

Reply:
Yeah for best optimization you'd want to some how gc this thing when the resulting stuff goes away, but how/when to invoke oncleanup seems hard to know it won't be referenced again